### PR TITLE
Create com.github.bitseather.weather.json

### PR DESCRIPTION
**Pull request by bil-elmoussaoui**
There's an issue with the appdata file, that's why the appstream-compose is set to false. Already reported upstream!

### CODE REVIEW

**Comment by Barthalion** [comment]
bot, build com.github.bitseater.weather

File: com.github.bitseather.weather.json [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+            "buildsystem": "meson",
+            "config-opts": ["--buildtype=release"],
+            "sources": [{
+                    "type": "archive",
+                    "url": "https://github.com/bitseater/weather/archive/0.5.3.tar.gz",
+                    "sha256": "36849f4220e75296610ebb469df243db3e36a1549dd184e4364e9b92282921f8"
+                },
+                {
+                    "type": "patch",
+                    "path": "meson-fix.patch"
```
Comment (Barthalion) [inline com.github.bitseather.weather.json]: Was it proposed upstream?

File: com.github.bitseather.weather.json [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+                "url": "https://github.com/GNOME/libgee/archive/0.20.0.tar.gz",
+                "sha256": "42fe6d651910cb8b720167f71c5255a1b7b1afc82fecd3f31e61f9602b3b1335"
+            }]
+        },
+        {
+            "name": "granite",
+            "buildsystem": "cmake-ninja",
+            "config-opts": [
+                "-DCMAKE_BUILD_TYPE=Release",
+                "-DCMAKE_INSTALL_LIBDIR=/app/lib"
```
Comment (TingPing) [inline com.github.bitseather.weather.json]: Setting libdir *should* no longer be needed.

File: com.github.bitseather.weather.json [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+            ],
+            "rm-configure": true,
+            "sources": [{
+                    "type": "archive",
+                    "url": "https://launchpad.net/libappindicator/12.10/12.10.0/+download/libappindicator-12.10.0.tar.gz",
+                    "sha256": "d5907c1f98084acf28fd19593cb70672caa0ca1cf82d747ba6f4830d4cc3b49f"
+                },
+                {
+                    "type": "patch",
+                    "path": "no-python.patch"
```
Comment (Barthalion) [inline com.github.bitseather.weather.json]: You can probably consider going more nuclear with this: https://github.com/flathub/io.github.Pithos/blob/master/appindicator-rip-build-system-apart.patch

File: meson-fix.patch [hunk 4]
```diff
@@ -0,0 +1,34 @@
+--- a/meson/post_install.py	2018-02-10 19:52:37.000000000 +0100
```
Comment (TingPing) [inline meson-fix.patch]: This was submitted upstream?

File: com.github.bitseather.weather.json [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+    "app-id": "com.github.bitseater.weather",
+    "runtime": "org.gnome.Platform",
+    "runtime-version": "3.26",
+    "sdk": "org.gnome.Sdk",
+    "command": "com.github.bitseater.weather",
+    "build-options": {
+        "cflags": "-O2  -Wno-deprecated-declarations",
+        "cxxflags": "-O2"
+    },
+    "appstream-compose": false,
```
Comment (Barthalion) [inline com.github.bitseather.weather.json]: Correct me if I'm wrong, but doesn't that mean that no appdata will be exported correctly?

File: meson-fix.patch (hunk 4 above)
Comment (bil-elmoussaoui) [reply meson-fix.patch]: Yes and was merged

**Comment by Barthalion** [comment]
Appid is different from manifest file name.

File: com.github.bitseather.weather.json (hunk 1 above)
Comment (bil-elmoussaoui) [reply com.github.bitseather.weather.json]: Yeah, and already merged along with some other fixes! Just waiting for a new release

File: com.github.bitseather.weather.json (hunk 3 above)
Comment (bil-elmoussaoui) [reply com.github.bitseather.weather.json]: I'm going to use this patch I guess..

File: com.github.bitseather.weather.json (hunk 5 above)
Comment (bil-elmoussaoui) [reply com.github.bitseather.weather.json]: Yeah, what I wanted is until a new release is released with the corrected appdata file, to verify if the build is successful on other architectures.

File: com.github.bitseather.weather.json (hunk 3 above)
Comment (bil-elmoussaoui) [reply com.github.bitseather.weather.json]: Just took a look at the patch, the patch removes also Vala bindings...Not sure if that will affect using AppIndicator with this app?

**Comment by bil-elmoussaoui** [comment]
I have updated the application to 0.5.4. Which includes the meson patch, and also fixes the issue with the appdata file. Can you guys trigger a new build so I can check if there's anything else that should be modified/added? Thanks!

**Comment by Barthalion** [comment]
bot, build com.github.bitseater.weather

**Comment by bil-elmoussaoui** [comment]
Is there anything left that should be modifies/updated?

**Comment by nedrichards** [comment]
Repository has been created: https://github.com/flathub/com.github.bitseater.weather

**Comment by bil-elmoussaoui** [comment]
@nedrichards Thanks! Can you add  @bitseater as a collaborator too?

**Comment by bitseater** [comment]
Thanks to all of you, especially @bil-elmoussaoui

**Comment by nedrichards** [comment]
Already done.

**Pull request closed without merging**